agent/agent_endpoint_test.go

```diff
@@ -6011,7 +6011,7 @@ func TestAgent_Monitor(t *testing.T) {
 			req = req.WithContext(cancelCtx)
 
 			resp := httptest.NewRecorder()
-			handler := a.srv.handler(true)
+			handler := a.srv.handler()
```

Comment:
Wouldn't this change the test behaviour as we are now passing the default value of `agent.config.EnableDebug` which is equivalent to having `a.srv.handler(false)` in the old implementation.

In all cases I think we need to add a test that cover that use case (`agent.config.EnableDebug = true`)

Reply:
Does CI run these UTs?

I think EnableDebug is not used in the test cases - they are just there because we had this in the handler argument.
If it had changed the test behaviours the test cases would have failed. Isnt' it?

Reply:
yes that's right. Can you please check if the behaviour is tested elsewhere?